optimize/cmaes_test.go

```diff
@@ -0,0 +1,131 @@
+// Copyright ©2017 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package optimize
+
+import (
+	"errors"
+	"math"
+	"testing"
+
+	"gonum.org/v1/gonum/optimize/functions"
+)
+
+func TestCmaEsChol(t *testing.T) {
+	for i, test := range []struct {
+		dim      int
+		problem  Problem
+		method   *CmaEsChol
+		settings *Settings
+		good     func(*Result, error) error
+	}{
+		{
```

Comment:
I think there should definitely be some test cases with several minima with: a) initial data (mean, variance) around the global minimum such that the algorithm is unlikely to escape to a local minimum b) initial data that cover several local minima and the global minimum such that the algorithm is very likely to find the global minimum.

Reply:
Done.